ctp2_code/ai/strategy/scheduler/scheduler.cpp

```diff
@@ -664,9 +664,6 @@ void Scheduler::Match_Resources(const bool move_armies)
 
 			AI_DPRINTF(k_DBG_SCHEDULER, m_playerId, goal_ptr->Get_Goal_Type(), -1, ("\n"));
 			AI_DPRINTF(k_DBG_SCHEDULER, m_playerId, goal_ptr->Get_Goal_Type(), -1, ("\n"));
-			// Assuming that the list is still sorted,
-			// and the following has only Goal::BAD_UTILITY
-			break;
```

Comment:
That isn't really a good idea to remove it without testing. If the assumption is false and the list isn't sorted anymore than we have a bug here, since the AI is supposed to consider the goals with higher priority first.

Anyway, if we leave the loop here, then committed_agents_test == committed_agents should be true, actually it should be true at any point in the loops, which doesn't seem to be true. In fact, it should updated whenever agents are committed or rolled back.

In fact, it is another optimization and the behavior is correct as long as committed_agents <= committed_agents_test.

Actual, you can have thousands of goals in m_goals, and it is quite some code that is executed during one iteration. Therefore, you want to optimize this as much as you can.

Reply:
I agree that this change is incorrect. Would be better if committed_agents_test could be calculated in such a way that the Assert could remain the same, as I can imagine there are valid and invalid ways for committed_agents to be < committed_agents_test?